lua/bufferline/icons.lua

```diff
@@ -45,53 +44,49 @@ local icons = {
     end
   end),
 
-  --- @param bufnr integer
-  --- @param buffer_status bufferline.buffer.activity.name
-  --- @return string icon, string highlight_group
-  get_icon = function(bufnr, buffer_status)
-    if ok == false then
-      utils.notify(
-        'barbar: bufferline.icons is set to v:true but "nvim-dev-icons" was not found.' ..
-          '\nbarbar: icons have been disabled. Set `bufferline.icons` to `false` or ' ..
-          'install "nvim-dev-icons" to disable this message.',
-        vim.log.levels.WARN
-      )
-
-      if type(vim.g.bufferline) == 'table' then
-        command('let g:bufferline.icons = v:false')
+  get_icon = ok and
```

Comment:
`get_icon` should be more performant now, because its behavior is determined at module load (depending on whether `nvim-web-devicons` is installed) rather than checking for `ok == false`, when `ok` will never change.